src/OrchardCore.Modules/Orchard.Users/Services/UserStore.cs

```diff
@@ -322,19 +342,41 @@ public Task<IList<string>> GetRolesAsync(User user, CancellationToken cancellati
             return Task.FromResult<IList<string>>(user.RoleNames);
         }
 
-        public Task<bool> IsInRoleAsync(User user, string roleName, CancellationToken cancellationToken)
+        public async Task<bool> IsInRoleAsync(User user, string normalizedRoleName, CancellationToken cancellationToken)
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
-            return Task.FromResult(user.RoleNames.Contains(roleName));
+            if (string.IsNullOrWhiteSpace(normalizedRoleName))
+            {
+                throw new ArgumentException("Value cannot be null or empty.", nameof(normalizedRoleName));
+            }
+
+            var roleNames = await _roleProvider.GetRoleNamesAsync();
+            var roleName = roleNames?.FirstOrDefault(r => _normalizeKey.Normalize(r) == normalizedRoleName);
+
+            return !string.IsNullOrWhiteSpace(roleName) && user.RoleNames.Contains(roleName);
         }
 
-        public Task<IList<User>> GetUsersInRoleAsync(string roleName, CancellationToken cancellationToken)
+        public async Task<IList<User>> GetUsersInRoleAsync(string normalizedRoleName, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            if (string.IsNullOrEmpty(normalizedRoleName))
+            {
+                throw new ArgumentNullException(nameof(normalizedRoleName));
+            }
+
+            var roleNames = await _roleProvider.GetRoleNamesAsync();
+            var roleName = roleNames?.FirstOrDefault(r => _normalizeKey.Normalize(r) == normalizedRoleName);
+
+            if (!string.IsNullOrWhiteSpace(roleName))
+            {
+                var users = await _session.Query<User>().ListAsync();
```

Comment:
That's very bad, you are loading all users in memory and then filter by role. If it's necessary then it would require a Reduced index instead.